src/commonMain/kotlin/me/alllex/parsus/token/RegexToken.kt

```diff
@@ -0,0 +1,71 @@
+package me.alllex.parsus.token
+
+import me.alllex.parsus.parser.GrammarContext
+
+/**
+ * A token that [matches] the input using a [regex].
+ */
+class RegexToken(
+    regex: Regex,
+    name: String? = null,
+    ignored: Boolean = false,
+) : Token(name, ignored) {
+
+    private val pattern: String = regex.pattern
+    private val regex: Regex = prependPatternWithInputStart(pattern, regex.options)
+//    private val matcher: Matcher = this.regex.toPattern().matcher("")
+    //private val matcher: RegexMatcher = this.regex.toMatcher()
+
+    private fun prependPatternWithInputStart(patternString: String, options: Set<RegexOption>): Regex {
+        return if (patternString.startsWith(inputStartPrefix))
+            patternString.toRegex(options)
+        else {
+//            val newlineAfterComments = if (RegexOption.COMMENTS in options) "\n" else ""
+//            val patternToEmbed = if (RegexOption.LITERAL in options) Regex.escape(patternString) else patternString
+//            ("$inputStartPrefix(?:$patternToEmbed$newlineAfterComments)").toRegex(options - RegexOption.LITERAL)
+          "$inputStartPrefix(?:$patternString)".toRegex(options)
+          // TODO is the warning correct?
+          //      > Unnecessary non-capturing group '(?:patternString)'
+        }
+    }
```

Comment:
`RegexOption.LITERAL` and `RegexOption.COMMENTS` are JVM only - so I simplified this code to get it working as a short-term fix.